src/midiEditorCommands.cpp

```diff
@@ -1121,6 +1121,59 @@ void postMidiChangePitch(int command) {
 	outputMessage(s);
 }
 
+void postMidiMoveStart(int command) {
+	HWND editor = MIDIEditor_GetActive();
+	MediaItem_Take* take = MIDIEditor_GetTake(editor);	
+	// Get selected notes.
+	vector<MidiNote> selectedNotes = getSelectedNotes(take);
+	auto count = selectedNotes.size();
+	if (count == 0) {
+		return;
+	}
+	auto firstStart = selectedNotes.cbegin()->start;
+	bool generalize = count >= 8 || any_of(
+		selectedNotes.begin(), selectedNotes.end(),
+		[firstStart](MidiNote n) { return firstStart != n.start; }
+		);
+	if (!generalize) {
+		previewNotes(take, selectedNotes);
+	}
+	if (shouldReportNotes) {
+		ostringstream s;
+		if (generalize) {
+			s << count << (count == 1 ? " note" : " notes ");
+			switch (command) {
+				case 40181:
+					s << "pixel left";
+					break;
+				case 40182:
+					s << "pixel right";
+					break;
+				case 40183:
+					s << "grid unit left";
+					break;
+				case 40184:
+					s << "grid unit right";
+					break;				
+				default:
+					s << "start moved";
+					break;
+			}
+		} else{ 
+			for (auto note = selectedNotes.cbegin(); note != selectedNotes.cend(); ++note) {
```

Comment:
Because we're now using C++ 17, you can probably just do this (but I haven't tested):
`for (auto note: as_const(selectedNotes)) {`
(You need to `#include <utility>` for as_const.)
However, this is just something to consider, not a required change for approval.

Reply:
I had tried something like that the other day, specifically:
`for (auto const& note: selectedNotes) {`
However, I haven't managed in converting the `if (note != selectedNotes.cend() - 1) {`: Tried
`if (note != *(selectedNotes.cend() - 1)) {` but then, it can't compare const MidiNote against const MidiNote. I assume I have to implement an operator for that?